fdbrpc/include/fdbrpc/ReplicationPolicy.h

```diff
@@ -159,8 +186,17 @@ struct PolicyAcross final : IReplicationPolicy {
 
 	const std::string& attributeKey() const { return _attribKey; }
 
+	void traverseForMaxDepth(int& globalMax) const override {
+		globalMax = std::max(globalMax, depth());
+		_maxdepth = globalMax;
+		_policy->traverseForMaxDepth(globalMax);
+	}
+
+	inline bool isSingleAcrossOverPolicyOne() const noexcept {
+		return maxdepth() == 2 && dynamic_cast<PolicyOne*>(_policy.getPtr());
```

Comment:
Are we simply checking if "_policy" is populated here? And, are we inferring that if "depth == 2" then "_policy" must be of type "PolicyOne"?

Reply:
And it is not possible to have a pattern like "PolicyAnd(...., PolicyOne)", or any other pattern that is not "PolicyAcross→PolicyOne" but still causes "maxdepth() == 2"?

Reply:
A related question here: is it safer to simply check if the current policy is of type "PolicyAcross" rather than doing a check based on "maxDepth()" and inferring that the current policy is "PolicyAcross" (if so, there might not be a need to cache "_depth" and "_maxDepth" at all) or is there a reason(s) why it is not correct/sufficient to do that check? Thanks!